### PR TITLE
Check and escape

### PR DESCRIPTION
**Pull request by scottnelle**
- Escape output when building settings form
- Check variables rather than suppressing errors

### CODE REVIEW

File: twitter_oauth/OAuth.php [hunk 1]
```diff
@@ -379,11 +381,11 @@ public function get_normalized_http_method() {
 	 */
 	public function get_normalized_http_url() {
 		$parts = parse_url( $this->http_url );
-
-		$port = @$parts['port'];
+! empty( $parts[''] ) ? $parts[''] : '';
```
Comment (mboynes) [inline twitter_oauth/OAuth.php]: rogue statement?

File: twitter_oauth/OAuth.php [hunk 2]
```diff
@@ -623,8 +638,13 @@ private function get_token( &$request, $consumer, $token_type="access" ) {
...
 	private function check_signature( &$request, $consumer, $token ) {
 		// this should probably be in a different method
-		$timestamp = @$request->get_parameter( 'oauth_timestamp' );
-		$nonce = @$request->get_parameter( 'oauth_nonce' );
+		if ( is_object( $request ) && method_exists( $request, 'get_parameter' ) ) {
+			$timestamp = $request->get_parameter( 'oauth_timestamp' );
+			$nonce = $request->get_parameter( 'oauth_nonce' );
+		} else {
+			$timestamp = NULL;
+			$nonce = NULL;
```
Comment (mboynes) [inline twitter_oauth/OAuth.php]: downcase all the `null`s

**Comment by mboynes** [comment]
:+1:

File: twitter_oauth/OAuth.php (hunk 1 above)
Comment (scottnelle) [inline twitter_oauth/OAuth.php]: woops, was writing a little template for myself and then didn't clean it up!

**Pull request merged**